examples/finding_best_open_source_embedding_model/best_embedding_model_rag_app.ipynb

```diff
@@ -0,0 +1,559 @@
+    "5. Evaluating each model's retrieval performance"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Environment Setup\n",
+    "\n",
+    "Before you begin, make sure to update the `OPENAI-API-KEY` in the `compose.yaml` file. Then execute the `install.sh` shell script to set up your environment. This script will:\n",
+    "\n",
+    "1. Install all necessary Docker containers required for the environment.\n",
+    "2. Load the dataset into the PostgreSQL Docker container, which will be used throughout this tutorial.\n",
+    "\n",
+    "**Note for macOS Users:**\n",
+    "  \n",
+    "To run the script, first make it executable by running the command: `chmod +x install.sh`. Then, execute the script with: `./install.sh`.\n",
+    "\n",
+    "Let's install the necessary Python libraries for this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install pandas python-dotenv psycopg2-binary Jinja2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dataset Ingestion\n",
+    "\n",
+    "Let's setup the PostgreSQL database and install [pgai](https://github.com/timescale/pgai)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os \n",
+    "import psycopg2\n",
+    "from dotenv import load_dotenv\n",
+    "\n",
+    "load_dotenv()\n",
+    "\n",
+    "DATABASE_CONNECTION_STRING = os.environ[\"DATABASE_CONNECTION_STRING\"]\n",
+    "\n",
+    "def connect_db():\n",
+    "    return psycopg2.connect(DATABASE_CONNECTION_STRING)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 34,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with connect_db() as conn:\n",
+    "    with conn.cursor() as cur:\n",
+    "        cur.execute(\"CREATE EXTENSION IF NOT EXISTS ai VERSION '0.5.0' CASCADE;\")\n",
+    "\n",
+    "    with conn.cursor() as cur:\n",
+    "        cur.execute(\"\"\"\n",
+    "            CREATE TABLE IF NOT EXISTS essays (\n",
+    "                id BIGINT PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,\n",
+    "                title TEXT NOT NULL,\n",
+    "                date TEXT,\n",
+    "                text TEXT NOT NULL\n",
+    "            );\n",
+    "          \"\"\") "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Loading the Dataset\n",
+    "\n",
+    "We'll now load Paul Graham's essays into our PostgreSQL database. The dataset consists of 215 essays with their titles, dates, and full text content. We'll verify the successful ingestion by displaying the first essay from the database."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with connect_db() as conn:\n",
```

Comment:
Instead of loading the data from a csv file (and leaving the user to figure that out), try this:

```
SELECT ai.load_dataset('sgoel9/paul_graham_essays', table_name => 'essays');
```

Reply:
Addendum:

```
SELECT ai.load_dataset('sgoel9/paul_graham_essays', table_name => 'essays', if_table_exists => 'append');
```

Reply:
👍 for using this but this then also needs pgai 0.6.0 not 0.5.0

Reply:
I actually couldn't find the function when using pgai 0.6.0 yesterday. @JamesGuthrie